lib/text/ui_text_displayer.js

```diff
@@ -644,30 +716,73 @@ shaka.text.UITextDisplayer = class {
       }
     }
 
-    style.lineHeight = cue.lineHeight;
+    const computedPositionAlign = this.computeCuePositionAlignment_(cue);
+    const computedPosition = this.computeCuePosition_(cue);
+
+    // The positionAlign attribute is an alignment for the text container in
+    // the dimension of the writing direction.
+    if (computedPositionAlign === Cue.positionAlign.LEFT) {
+      style.alignItems = 'flex-start';
+    } else if (computedPositionAlign === Cue.positionAlign.RIGHT) {
+      style.alignItems = 'flex-end';
+    }
 
     // The position defines the indent of the text container in the
     // direction defined by the writing direction.
-    if (cue.position != null) {
-      if (cue.writingMode == Cue.writingMode.HORIZONTAL_TOP_TO_BOTTOM) {
-        style.paddingLeft = cue.position;
+    // TODO: factorize code since width/height, left/top and right/bottom
+    //       have same role, depending on the writing mode
+    if (cue.writingMode === Cue.writingMode.HORIZONTAL_TOP_TO_BOTTOM) {
+      // Horizontal writing
+      if (computedPositionAlign === Cue.positionAlign.LEFT) {
+        // Left side of cue box is positioned at 'computedPosition'
+        style.paddingLeft = computedPosition + '%';
+        style.width = (100 - computedPosition) + '%';
```

Comment:
Is padding and width really the right way to align the cue box?  Padding is internal to the element, and width includes padding.

So if the element has a solid background, that background covers the padding area, too.

Also, since width includes padding, if computedPosition is 50%, you end up with width=50%, padding=50%.  If width == padding, there's no room for content, right?

Reply:
This padding is not applied to the element containing the text, but to the container of this element.
So, for example, on the image below, you can see a padding of 50% *and* a width of 50%:
![image](https://user-images.githubusercontent.com/10497009/223671717-25a775fc-680e-4e51-aeeb-aa5e79daa855.png)